test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,552 @@ describe("BaseAxis", () => {
+              TestMethods.numAttr(firstAnnotationRect, "height"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("moves rectangles that would overlap to different rows leftwards (orientation: left)", () => {
+            let axis = new Plottable.Axis(scale, "left");
+            axis.annotatedTicks([50, 51]);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "x") - TestMethods.numAttr(secondAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+
+          it("moves rectangles that would overlap to different rows rightwards (orientation: right)", () => {
+            let axis = new Plottable.Axis(scale, "right");
+            axis.annotatedTicks([50, 51]);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            let firstAnnotationRect = d3.select(annotationRects[0][0]);
+            let secondAnnotationRect = d3.select(annotationRects[0][1]);
+            assert.strictEqual(TestMethods.numAttr(secondAnnotationRect, "x") - TestMethods.numAttr(firstAnnotationRect, "x"),
+              TestMethods.numAttr(firstAnnotationRect, "width"), "rectangle offset by previous rectangle");
+            svg.remove();
+          });
+        });
+
+      });
+
+      it("hides annotations if rectangles are outside the margin area", () => {
+        let mockScale = new Plottable.Scales.Linear;
+        let annotatedTicks = [50, 51, 150];
+        mockScale.domain(annotatedTicks);
+
+        let axis = new Plottable.Axis(mockScale, "bottom");
+        axis.annotatedTicks(annotatedTicks);
+        axis.annotationsEnabled(true);
+        axis.margin(25);
+
+        let svg = TestMethods.generateSVG(300, 300);
+        axis.renderTo(svg);
+
+        axis.content().selectAll(".annotation-rect").each(function() {
+          let annotationRect = d3.select(this);
+          let bbox = this.getBBox();
+          let insideMargin = bbox.x >= 0 && bbox.x + bbox.width <= axis.width() &&
+                             bbox.y >= axis.height() - axis.margin() && bbox.y + bbox.height <= axis.height();
+          let visibilityAttr = insideMargin ? "visible" : "hidden";
+          assert.strictEqual(annotationRect.attr("visibility"), visibilityAttr, "annotation rect inside margin area");
```

Comment:
The other advantage of a split `assert` is that we can provide a better assert-message; with this line, we instead get a message like this:
```
AssertionError: annotation rect inside margin area: expected 'hidden' to equal 'visible'
```
---
Maybe you could reword the assert-message instead? Right now it's hard to understand what `"hidden"`/`"visible"` have to do with `"annotation rect inside margin area"`. Something like `"annotations that don't fit are hidden"` might be better, but then you can still get the following case:
```
AssertionError: annotations that don't fit are hidden: expected 'hidden' to equal 'visible'
```
which is really confusing unless you realize that the particular annotation being tested is supposed to be visible.